msal/src/main/java/com/microsoft/identity/client/internal/api/BrokerClientIdRefreshTokenAccessor.java

```diff
@@ -49,6 +50,25 @@
 
     private static final String TAG = BrokerClientIdRefreshTokenAccessor.class.getSimpleName();
 
+    private static void throwIfNotValidBroker(final Context context) throws MsalClientException {
+        final BrokerValidator brokerValidator = new BrokerValidator(context);
+        final Set<BrokerData> validBrokers = brokerValidator.getValidBrokers();
+
+        for (final BrokerData brokerData : validBrokers) {
+            if (brokerData.packageName.equals(context.getPackageName())) {
+                if (!brokerValidator.verifySignature(context.getPackageName())) {
+                    throw new MsalClientException(NOT_ELIGIBLE_TO_USE_BROKER, "This can only be invoked by Broker apps with a valid signature hash.");
```

Comment:
>NOT_ELIGIBLE_TO_USE_BROKER [](start = 50, length = 26)

nit: shouldn't this be something like "INVALID_BROKER_SIGNATURE_HASH" and the below one be "INVALID_BROKER_APP"

Reply:
This comment is no longer valid as I've refactored the code...but the idea is that at this layer we just know if the broker is valid or not...we don't know why...resolving comment for now